proteus/MeshTools.py

```diff
@@ -3734,6 +3740,29 @@ def computeGeometricInfo(self):
         for m in self.meshList:
             m.computeGeometricInfo()
 
+def buildReferenceSimplex(nd=2):
+    """
+    This function creates and returns a Proteus mesh object for the reference 
+    simplex.  This function can helpful for test development.
+    """
+    from proteus import Domain
+    from proteus import TriangleTools
+    if nd == 2:
+        unit_simplex_domain_2d = Domain.unitSimplex(2)
+        polyfile = "reference_triangle"
+        unit_simplex_domain_2d.writePoly(polyfile)
+        tmesh = TriangleTools.TriangleBaseMesh(baseFlags="Yp",
+                                               nbase=1,
+                                               verbose=False)
+        tmesh.readFromPolyFile(polyfile)
+        mesh = tmesh.convertToProteusMesh(verbose=0)
+        mesh.partitionMesh()
+        mesh.globalMesh = mesh
+        return mesh
+    if nd == 3:
+        pass
```

Comment:
You implemented Domain.unitSimplex(3) above so I guess the 3D should be just ripping out the code from NumericalSolution that calls tetgen on a 3d polyfile.